pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1444,7 +1466,12 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(ctx context.Context, clusterName
 		if err != nil && err != ErrNotFound {
 			return err
 		}
-		if floatingIP != nil {
+
```

Comment:
Please do it like ：
>>
	keepFloatingAnnotation, err := getBoolFromServiceAnnotation(service, ServiceAnnotationLoadBalancerKeepFloatingIP, false)
	if err != nil {
		return err
	}

	if !keepFloatingAnnotation {
		if loadbalancer.VipPortID != "" {
			portID := loadbalancer.VipPortID
			floatingIP, err := getFloatingIPByPortID(lbaas.network, portID)
			if err != nil && err != ErrNotFound {
				return err
			}
			if floatingIP != nil {
				err = floatingips.Delete(lbaas.network, floatingIP.ID).ExtractErr()
				if err != nil && !isNotFound(err) {
					return err
				}
			}
		}
	}